src/webgpu/util/math.ts

```diff
@@ -61,26 +72,63 @@ export function isSubnormalScalar(val: Scalar): boolean {
 }
 
 /** Utility to pass TS numbers into |isSubnormalNumber| */
-export function isSubnormalNumber(val: number): boolean {
+export function isSubnormalNumberF32(val: number): boolean {
   return val > kValue.f32.negative.max && val < kValue.f32.positive.min;
 }
 
 /** @returns if number is in the finite range of f32 */
-export function isF32Finite(n: number) {
+export function isFiniteF32(n: number) {
   return n >= kValue.f32.negative.min && n <= kValue.f32.positive.max;
 }
 
+/** @returns 0 if |val| is a subnormal f16 number, otherwise returns |val| */
+export function flushSubnormalNumberF16(val: number): number {
+  return isSubnormalNumberF16(val) ? 0 : val;
+}
+
+/** @returns 0 if |val| is a subnormal f16 number, otherwise returns |val| */
+export function flushSubnormalScalarF16(val: Scalar): Scalar {
+  return isSubnormalScalarF16(val) ? f16(0) : val;
+}
+
 /**
- * @returns the next single precision floating point value after |val|,
+ * @returns true if |val| is a subnormal f16 number, otherwise returns false
+ * 0 is considered a non-subnormal number by this function.
+ */
+export function isSubnormalScalarF16(val: Scalar): boolean {
+  if (val.type.kind !== 'f16') {
+    return false;
+  }
+
+  if (val === f16(0)) {
+    return false;
+  }
+
+  const u16_val = new Uint16Array(new Float16Array([val.value.valueOf() as number]).buffer)[0];
+  return (u16_val & 0x7f800000) === 0;
+}
+
+/** Utility to pass TS numbers into |isSubnormalNumber| */
```

Comment:
Not sure I understand this comment as we don't call `isSubnormalNumber` in here?